data/transactions/logic/eval.go

```diff
@@ -799,6 +825,16 @@ func (cx *EvalContext) remainingBudget() int {
 	if cx.runModeFlags == runModeSignature {
 		return int(cx.Proto.LogicSigMaxCost) - cx.cost
 	}
+
+	// restrict clear state programs from using more than standard unpooled budget
+	// cx.Txn is not set during check()
```

Comment:
Unit tests!  It's nil during `check()`, never during `eval()`. It would be annoying to fill it, since the public interface `CheckProgram()` does not take the transaction index.  It would probably be possible to add it, but I didn't want to change calling code.

Reply:
@jannotti Thanks for the reply, I think I'm following better now.  Resolving.